pkg/controller/workloadspread/workloadspread_controller.go

```diff
@@ -327,16 +328,49 @@ func groupPod(ws *appsv1alpha1.WorkloadSpread, pods []*corev1.Pod) map[string][]
 	for i := range pods {
 		injectWS := getInjectWorkloadSpreadFromPod(pods[i])
 		if injectWS == nil || injectWS.Name != ws.Name {
-			continue
-		}
-		if _, exist := podMap[injectWS.Subset]; exist {
+			// process the pods that were created before workloadSpread
+			if matchedSubset, err := r.getSuitableSubsetForOldPod(ws, pods[i]); err != nil {
+				klog.Errorf("failed to get a suitable subset for the old pod (%s/%s) which was created before workloadSpread (%s/%s), err: %v",
+					pods[i].Namespace, pods[i].Name, ws.Namespace, ws.Name, err)
+			} else if matchedSubset != nil {
+				podMap[matchedSubset.Name] = append(podMap[matchedSubset.Name], pods[i])
+			}
+		} else if _, exist := podMap[injectWS.Subset]; exist {
 			podMap[injectWS.Subset] = append(podMap[injectWS.Subset], pods[i])
 		}
 	}
 
 	return podMap
 }
 
+// getSuitableSubsetForOldPod returns an suitable subset for the pod that was created before workloadSpread.
+// If there is no suitable subset for the pod, this function will return a empty pointer.
+func (r *ReconcileWorkloadSpread) getSuitableSubsetForOldPod(ws *appsv1alpha1.WorkloadSpread, pod *corev1.Pod) (*appsv1alpha1.WorkloadSpreadSubset, error) {
+	var err error
+	var node *corev1.Node
+
+	if len(pod.Spec.NodeName) > 0 {
+		node = &corev1.Node{}
+		err = r.Get(context.TODO(), types.NamespacedName{Name: pod.Spec.NodeName}, node)
+	}
+
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, err
+	} else if err == nil && node != nil {
+		for i := range ws.Spec.Subsets {
+			subset := &ws.Spec.Subsets[i]
+			// in case of that this pod was scheduled to the node which matches a subset of workloadSpread
+			if matched, err := nodeMatchesSubset(node, subset); err != nil {
+				return nil, err
+			} else if matched {
```

Comment:
shall we also check the subset patch ? One can have subset that only have patch but without any affinity or toleration changes. maybe we should leave it as a TODO

Reply:
done

Reply:
I left it as a TODO